deps.ffmpeg: Enable PDBs for FFmpeg builds on Windows

RytoEX opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst -->

### Description
<!--- Describe your changes in detail. -->
<!--- If this change includes UI elements, please include screenshots. -->

Enable PDBs for FFmpeg builds on Windows if the Configuration is set to Debug or RelWithDebInfo.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open GitHub Issue, or implements feature request -->
<!--- from the Ideas page, please link to the issue here. -->

Having PDBs for FFmpeg can be useful for debugging crashes. This was one of the original goals of moving the FFmpeg builds to be natively compiled.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment (hardware, OS version, etc.),-->
<!--- and the tests you ran, including how it may affect other areas of code. -->

Tested locally about a year ago. Should probably be retested.

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
<!--- - Bug fix (non-breaking change which fixes an issue) -->
<!--- - New feature (non-breaking change which adds functionality) -->
- Tweak (non-breaking change to improve existing functionality)
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
<!--- - Code cleanup (non-breaking change which makes code smaller or more readable) -->
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
<!--- - Documentation (a change to documentation pages) -->

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

PatTheMav commented on deps.ffmpeg/99-ffmpeg.ps1:
```diff
@@ -67,7 +67,7 @@ function Configure {
         ('--extra-cflags=' + "'-D_WINDLL -MD -D_WIN32_WINNT=0x0A00" + $(if ( $Target -eq 'arm64' ) { ' -D__ARM_PCS_VFP' }) + "'")
         ('--extra-cxxflags=' + "'-MD -D_WIN32_WINNT=0x0A00'")
-        ('--extra-ldflags=' + "'-APPCONTAINER:NO -MACHINE:${Target}'")
+        ('--extra-ldflags=' + "'-APPCONTAINER:NO -MACHINE:${Target}'" + $(if ( $Configuration -match '(Debug|RelWithDebInfo)' ) { ' -DEBUG' }))
```
Wouldn't it make sense to always add this flag? Calling PDBs "debug"-symbols is a bit of a misleading name, as "debug" is their purpose but is not tied to the build configuration. It is perfectly fine (and indeed expected) to generate PDB symbol files in `Release` configuration (or _particularly_ for that configuration) to ship stripped release binaries to users while still being able to debug/symbolicate issues.

RytoEX replied:
We could. My recollection from our conversation when I first wrote this a year ago was that blanket adding `-DEBUG` all the time was a quick hack and that we should only add it as needed. If our current consensus is that we should always use this flag, then we can do that.

RytoEX replied:
The default configuration for `.github/actions/build-ffmpeg/action.yaml` is `Release` and it seems that we currently specify that in `.github/workflows/main.yaml` and `.github/workflows/scheduled.yaml`.

PatTheMav replied:
That was my hunch - so if we actually want to have PDBs generated for `obs-deps` releases, then the `DEBUG` switch needs to be probably added regardless of selected build configuration.

This will work for CMake-based projects using the MSVC build system, but I'm not sure about custom build systems like FFmpeg's.

RytoEX replied:
How do we want to proceed here?

RytoEX replied:
I've pushed an implementation as [written above](https://github.com/obsproject/obs-deps/pull/280#discussion_r2096576063):
```pwsh
        ('--extra-ldflags=' + "'-APPCONTAINER:NO -MACHINE:${Target} -DEBUG" +
            $(if ( $Configuration -match '(Release|RelWithDebInfo|MinSizeRel)' ) { ' -OPT:REF -OPT:ICF -LTCG -INCREMENTAL:NO' }) + "'")
```

If we want to expand this to `Setup-BuildParameters`, we can make that change later.

RytoEX committed:
deps.ffmpeg: Enable PDBs for FFmpeg builds on Windows

Enable PDBs for FFmpeg builds on Windows for all configurations. If the
configuration is set to Release, or RelWithDebInfo, or MinSizeRel,
adjust the linker options accordingly.

RytoEX committed:
deps.ffmpeg: Enable PDBs for FFmpeg builds on Windows

Enable PDBs for FFmpeg builds on Windows for all configurations. If the
configuration is set to Release, or RelWithDebInfo, or MinSizeRel,
adjust the linker options accordingly.

RytoEX commented:
Just confirming that the recent push still creates PDBs for the DLLs:
<img width="623" height="202" alt="image" src="https://github.com/user-attachments/assets/77a52abc-c03d-450e-baa8-9ee2c628e78d" />

Executables are suffixed for some reason:
<img width="612" height="164" alt="image" src="https://github.com/user-attachments/assets/50d7d16f-7d47-411e-9f0b-8b28d66fa9ce" />

The PDBs are not currently copied/installed to the install dir.

RytoEX commented:
Apparently, I had forgotten to actually push the changes.

RytoEX committed:
deps.ffmpeg: Copy PDBs to the install dir

The make install command does not copy the PDBs to the install dir, so
we have to copy them manually.

PatTheMav commented on deps.ffmpeg/99-ffmpeg.ps1:
```diff
@@ -165,4 +166,9 @@ function Install {
     $Backup.GetEnumerator() | ForEach-Object { Set-Item -Path "env:\$($_.Key)" -Value $_.Value }
+
+    Push-Location "build_${Target}"
+    Copy-Item "ffmpeg_g.exe","ffprobe_g.exe" "$($script:ConfigData.OutputPath)/bin"
```
It seems that I chose to use single quotes for plain string tokens and only used double quotes when interpolation is necessary.

I dunno if the code is consistent that way, but at least the lines in `Build-Dependencies.ps1` suggest that pattern is in use.

RytoEX replied:
I don't think our code is consistent about it. The documentation for `Copy-Item` uses double quotes for nearly everything, and I was referencing that at the time. I can change it.
